temporal-sdk/src/main/java/io/temporal/client/WorkflowStubImpl.java

```diff
@@ -131,45 +131,71 @@ public WorkflowExecution start(Object... args) {
   }
 
   @Override
-  public <R> WorkflowUpdateHandle<R> updateWithStart(
-      UpdateWithStartWorkflowOperation<R> updateOperation, Object... startArgs) {
+  public <R> WorkflowUpdateHandle<R> startUpdateWithStart(
+      @Nonnull UpdateOptions<R> updateOptions,
+      Object[] updateArgs,
+      @Nonnull WithStartWorkflowOperation<?> startOp) {
     if (options == null) {
-      throw new IllegalStateException("Required parameter WorkflowOptions is missing");
+      throw new IllegalStateException(
+          "Required parameter WorkflowOptions is missing in WorkflowStub");
     }
-    if (!updateOperation.markInvoked()) {
-      throw new IllegalStateException("UpdateWithStartWorkflowOperation was already executed");
+    if (options.getWorkflowIdConflictPolicy() == null) {
+      throw new IllegalStateException(
+          "Required parameter WorkflowIdConflictPolicy in WorkflowOptions is missing in WorkflowStub");
     }
-    checkExecutionIsNotStarted();
+    if (!startOp.markInvoked()) {
+      throw new IllegalStateException("WithStartWorkflowOperation was already executed");
+    }
+    if (startOp.getStub() == null) {
+      throw new IllegalStateException(
+          "WithStartWorkflowOperation was created with different WorkflowStub");
+    }
```

Comment:
So this is worth discussing: the Start Operation's `getResult` needs a class for the Workflow's result type to work. When using the typed stub to initialize it, it is inferred. When using the untyped, it is provided by the user explicitly. However, when initialize it via the typed stub _and_ using the untyped stub to run startUpdateWithStart, the type inference step was skipped and any call to `startOp.getResult()` fails. I don't see a way around this other than removing the `getResult` shortcut entirely. Then this check can be safely removed.

Reply:
I think it is worth considering if the untyped stub `startUpdateWithStart` API should even take a `WithStartWorkflowOperation` ?

Reply:
I guess `startUpdateWithStart ` is supposed to take a `vargs` , and you can't have two varags

Reply:
Yup, @dandavison and I had that conversation and we were leaning towards using the `WithStartWorkflowOperation` for the untyped `startUpdateWithStart` API for these reasons:

(1) a signature like `startUpdateWithStart(UpdateOptions<R> options, Object[] updateArgs, Object[] startArgs)` is potentially dangerous because users might mix up the order of the args (counterpoint: the existing `signalWithStart` has the same problem)
(2) consistent use of the `startOp` parameter across typed and untyped APIs
(3) allows convenience function `getResult()`

(1) is the main reason, (2) and (3) are minor reasons.

Having said that, neither of us - so far - felt very strongly about it.